src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/begrunnelser/FinnGyldigeBegrunnelserForPeriodeContext.kt

```diff
@@ -0,0 +1,213 @@
+        when (standardBegrunnelse.vedtakBegrunnelseType) {
+            VedtakBegrunnelseType.REDUKSJON,
+            VedtakBegrunnelseType.EØS_INNVILGET,
+            VedtakBegrunnelseType.AVSLAG,
+            VedtakBegrunnelseType.ENDRET_UTBETALING,
+            VedtakBegrunnelseType.INNVILGET -> finnPersonerMedVilkårResultaterSomGjelderIPeriode()
+
+            VedtakBegrunnelseType.EØS_OPPHØR,
+            VedtakBegrunnelseType.ETTER_ENDRET_UTBETALING,
+            VedtakBegrunnelseType.OPPHØR -> finnPersonerMedVilkårResultaterSomGjelderRettFørPeriode()
+
+            VedtakBegrunnelseType.FORTSATT_INNVILGET -> throw Feil("FORTSATT_INNVILGET skal være filtrert bort.")
+        }
+
+    private fun finnPersonerMedVilkårResultaterSomGjelderIPeriode(): Map<Person, List<VilkårResultat>> =
+
+        personResultater.tilFørskjøvetVilkårResultatTidslinjeMap(persongrunnlag)
+            .mapNotNull { (aktør, vilkårResultatTidslinjeForPerson) ->
+                val person =
+                    persongrunnlag.personer.find { it.aktør.aktivFødselsnummer() == aktør.aktivFødselsnummer() }
+                val forskøvedeVilkårResultaterMedSammeFom =
+                    vilkårResultatTidslinjeForPerson.tilPerioderIkkeNull().singleOrNull {
+                        it.fom == vedtaksperiode.fom
+                    }?.verdi
+                if (person != null && forskøvedeVilkårResultaterMedSammeFom != null) {
+                    Pair(person, forskøvedeVilkårResultaterMedSammeFom)
+                } else {
+                    null
+                }
+            }.toMap().filterValues { it.isNotEmpty() }
+
+    private fun finnPersonerMedVilkårResultaterSomGjelderRettFørPeriode(): Map<Person, List<VilkårResultat>> =
+        personResultater.tilFørskjøvetVilkårResultatTidslinjeMap(persongrunnlag).mapNotNull { (aktør, tidsjlinje) ->
+            val person = persongrunnlag.personer.find { it.aktør.aktivFødselsnummer() == aktør.aktivFødselsnummer() }
+            val forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode =
+                tidsjlinje.tilPerioderIkkeNull().singleOrNull {
+                    it.tom?.plusDays(1) == vedtaksperiode.fom
+                }?.verdi
+
+            if (person != null && forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode != null) {
+                Pair(person, forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode)
+            } else {
+                null
+            }
+        }.toMap().filterValues { it.isNotEmpty() }
+
+    private fun Map<Person, List<VilkårResultat>>.filtrerPersonerUtenUtbetalingVedInnvilget(vedtakBegrunnelseType: VedtakBegrunnelseType) =
+        this.filterKeys {
+            if (vedtakBegrunnelseType == VedtakBegrunnelseType.INNVILGET) {
+                aktørIderMedUtbetaling.contains(it.aktør.aktørId) || it.type == PersonType.SØKER
+            } else {
+                true
+            }
```

Comment:
Hadde dette vært lettere å lese om det var
```suggestion
vedtakBegrunnelseType != VedtakBegrunnelseType.INNVILGET || 
  aktørIderMedUtbetaling.contains(it.aktør.aktørId) || 
  it.type == PersonType.SØKER
```
?